### PR TITLE
[DEVX] Update documentation

### PR DESCRIPTION
**Pull request by kml**
This PR updates documentation - README.md files, but also documentation in the code (comments).

### CODE REVIEW

File: distrib-core/lib/distrib_core/leader/watchdog.rb [hunk 1]
```diff
@@ -1,5 +1,3 @@
-require 'rainbow/refinement'
```
Comment (kml) [inline distrib-core/lib/distrib_core/leader/watchdog.rb:1]: This is a leftover. We don't use rainbow, we also don't specify it as a dependency of the gem.

File: rspec-distrib/README.md [hunk 2]
```diff
@@ -209,25 +254,19 @@ RSpec::Distrib.configure do |config|
...
 
-Specify timeout per spec file. An object that responds to `call` and receives
+To specify timeout per spec file use and object that responds to `call` and receives
 the spec file path as an argument. The proc returns the timeout in seconds.
+This is also useful for cases where some specs have a timeout strategy and some
+don't.
 
 ```ruby
 RSpec::Distrib.configure do |config|
-  config.timeout_proc = ->(spec_file) do
```
Comment (kml) [inline rspec-distrib/README.md:217]: `timeout_proc` is no longer supported. Current solution is to define callable `test_timeout`.

File: distrib-core/lib/distrib_core/configuration.rb [hunk 3]
```diff
@@ -4,12 +4,12 @@
...
 
 module DistribCore
-  # This module contains shared attrs instantiated by specific configuration classes.
+  # This module contains shared attributes instantiated by specific configuration classes.
   #
   # @see DistribCore::Distrib#configure
   module Configuration
     class << self
-      # Set global configuration. Can be set only one time
+      # Set global configuration. Can be set only one time.
```
Comment (AlexDavidoiu) [inline distrib-core/lib/distrib_core/configuration.rb:12]:
```suggestion
      # Set global configuration. Can be set only once.
```

File: distrib-core/lib/distrib_core/configuration.rb [hunk 4]
```diff
@@ -42,6 +47,8 @@ def current
 
     attr_writer :logger
 
+    # Set timeout for tests. It can be a number of seconds or a callable object that returns a number of seconds.
```
Comment (AlexDavidoiu) [inline distrib-core/lib/distrib_core/configuration.rb:50]: is there a particular reason as to why we use `Set` (and the rest of the verbs) and not `Sets`?

File: rspec-distrib/README.md [hunk 5]
```diff
@@ -111,6 +105,55 @@ the execution results `report_file` back to the leader.
 
 ![Worker workflow](docs/worker.png)
 
+## Development
+
+To start up development of the gem first make sure you could run following commands without problems:
```
Comment (AlexDavidoiu) [inline rspec-distrib/README.md:110]:
```suggestion
To start up development of the gem first make sure you could run the following commands without problems:
```

File: rspec-distrib/README.md [hunk 6]
```diff
@@ -111,6 +105,55 @@ the execution results `report_file` back to the leader.
...
+bundle install
+bundle exec rspec spec
+bundle exec rspec features
+bundle exec rubocop
+bundle exec yardoc --fail-on-warning
+```
+
+Tests under `spec` are unit tests, while `features` are integration tests.
+
+Features could be used for manuall tests while developing.
```
Comment (AlexDavidoiu) [inline rspec-distrib/README.md:122]:
```suggestion
Features could be used for manual tests while developing.
```

File: rspec-distrib/README.md [hunk 7]
```diff
@@ -302,9 +341,13 @@ they crash.
...
 
-[DRb]: https://ruby-doc.org/stdlib-2.5.3/libdoc/drb/rdoc/DRb.html
-[knapsack]: https://github.com/ArturT/knapsack
-[parallel_tests]: https://github.com/grosser/parallel_tests
+> What's the order of serving the spec files?
+
+Order is controlled by `tests_provider`. Default provided strategy is at [rspec/distrib/leader/tests_provider.rb](rspec-distrib/lib/rspec/distrib/leader/tests_provider.rb) - it's simplified one.
+
+In more advanced scenario spec files could be served from the slowest (basing on previous builds results) to
+fastest to reduce worker idle time, and reduce the risk of waiting for a long
```
Comment (AlexDavidoiu) [inline rspec-distrib/README.md:349]:
```suggestion
the fastest to reduce worker idle time, and reduce the risk of waiting for a long
```

**Review by AlexDavidoiu — APPROVED** [review]

**Review by ebeagusamuel — APPROVED** [review]
Nice work :heart:

File: distrib-core/lib/distrib_core/configuration.rb (hunk 4 above)
Comment (kml) [reply distrib-core/lib/distrib_core/configuration.rb:50]: No particular reason. Documentation in many places was already like that.

**Pull request merged**